test/e2e/networking.go

```diff
@@ -102,160 +103,245 @@ var _ = Describe("Networking", func() {
-		for i := 0; !passed && timeout.After(time.Now()); i++ {
-			time.Sleep(2 * time.Second)
-			Logf("About to make a proxy status call")
-			start := time.Now()
-			body, err = getStatus()
-			Logf("Proxy status call returned in %v", time.Since(start))
 			if err != nil {
-				Logf("Attempt %v: service/pod still starting. (error: '%v')", i, err)
-				continue
+				Failf("unable to create test service named [%s] %v on port %v", svc.Name, err, port)
+			} else {
+				Logf("Created service successfully [%s]", svc.Name)
 			}
-			// Finally, we pass/fail the test based on if the container's response body, as to whether or not it was able to find peers.
-			switch {
-			case string(body) == "pass":
-				Logf("Passed on attempt %v. Cleaning up.", i)
-				passed = true
-			case string(body) == "running":
-				Logf("Attempt %v: test still running", i)
-			case string(body) == "fail":
-				if body, err = getDetails(); err != nil {
-					Failf("Failed on attempt %v. Cleaning up. Error reading details: %v", i, err)
-				} else {
-					Failf("Failed on attempt %v. Cleaning up. Details:\n%s", i, string(body))
+
+			// Clean up service
+			defer func() {
+				By("Cleaning up the service")
+				if err := f.Client.Services(f.Namespace.Name).Delete(svc.Name); err != nil {
+					Failf("unable to delete svc %v: %v", svc.Name, err)
 				}
-			case strings.Contains(string(body), "no endpoints available"):
-				Logf("Attempt %v: waiting on service/endpoints", i)
-			default:
-				Logf("Unexpected response:\n%s", body)
-			}
-		}
+			}()
+
+			Logf("launching pod per node.....")
+			podNames := launchNetTestPodPerNode(port, nettestVersion, f, nodes, svcname)
+			// Clean up the pods
+			defer func() {
+				By("Cleaning up the webserver pods")
+				for _, podName := range podNames {
+					if err = f.Client.Pods(f.Namespace.Name).Delete(podName, nil); err != nil {
+						Logf("Failed to delete pod %s: %v", podName, err)
+					}
+				}
+			}()
+			Logf("Launched test pods for %v", port)
+			By("waiting for all of the webserver pods to transition to Running + reaching the Passing state.")
 
-		if !passed {
-			if body, err = getDetails(); err != nil {
-				Failf("Timed out. Cleaning up. Error reading details: %v", err)
-			} else {
-				Failf("Timed out. Cleaning up. Details:\n%s", string(body))
+			for _, podName := range podNames {
+				err = f.WaitForPodRunning(podName)
+				Expect(err).NotTo(HaveOccurred())
+				By(fmt.Sprintf("waiting for connectivity to be verified [ pod = %v ,  port =  %v ] ", podName, port))
+				//once response OK, evaluate response body for pass/fail.
+				passedPolling := pollPeerStatus(f, svc, timeoutSeconds)
+				if !passedPolling {
+					Failf("Failed at polling %v %v", podName, port)
+				}
 			}
-		}
-		Expect(string(body)).To(Equal("pass"))
-	})
+			Logf("Finished test pods for %v", port)
 
-})
+			//Once the polling for a service is completed,
+			//we send failure/success to the completion channel.
+			completionChannel <- svcname
```

Comment:
unless you're planning to actually use the service name (which you aren't currently?) a waitgroup still makes more sense here, imo.